topostats/grains.py

```diff
@@ -190,9 +190,14 @@ def find_grains(self):
         self.get_mask()
         self.tidy_border()
         self.label_regions(self.images["tidied_border"])
-        self.calc_minimum_grain_size(image=self.images["labelled_regions"])
-        self.remove_small_objects()
-        self.label_regions(self.images["objects_removed"])
-        self.get_region_properties()
-        self.colour_regions()
-        self.get_bounding_boxes()
+        try:
```

Comment:
Pythonic, but using try catch rather than pre-empting the error always makes me feel uncomfortable! Looks good.

Reply:
I guess 194 is going to actually throw the error and, if so, we don't want 195-199 to execute so they're included in the block.

Reply:
My understanding is that its "Pythonic" to use `try: ... except:" and is captured in the [EAPF principle](https://docs.python.org/3.5/glossary.html#term-eafp) "_Easier to ask for forgiveness than permission_" than the contrasting "_Look Before You Leap_" that is characterised by `if: ... else:`.

And spot on, if we don't find any grains then we don't want the remaining steps to be run.